### PR TITLE
New resource for Service Fabric Managed Clusters service

### PR DESCRIPTION
**Pull request by koikonom**

### CODE REVIEW

**Comment by koikonom** [comment]
image

File: website/docs/r/service_fabric_managed_cluster.html.markdown [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+* `name` - (Required) The name which should be used for this Resource Group. Changing this forces a new Resource Group to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Resource Group should exist.
+
+---
+
+* `authentication` - (Optional) Controls how connections to the cluster are authenticated. A `authentication` block as defined below.
+
+* `backup_restore_service` - (Optional) If true, backup service is enabled.
```
Comment (katbyte) [inline website/docs/r/service_fabric_managed_cluster.html.markdown:68]: this is not very clear from the name, should we make it
```suggestion
* `backup_and_restore_service_enabled` - (Optional) If true, backup service is enabled.
```

or even
```suggestion
* `backup_service_enabled` - (Optional) If true, backup service is enabled.
```

File: website/docs/r/service_fabric_managed_cluster.html.markdown [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+A `certificates` block supports the following:
+
+* `store` - (Required) The certificate store on the Virtual Machine to which the certificate should be added.
+
+* `url` - (Required) The URL of a certificate that has been uploaded to Key Vault as a secret
+
+---
+
+A `custom_fabric_setting` block supports the following:
```
Comment (katbyte) [inline website/docs/r/service_fabric_managed_cluster.html.markdown:126]: setting seems redundant
```suggestion
A `custom_fabric` block supports the following:
```

File: website/docs/r/service_fabric_managed_cluster.html.markdown [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+* `protocol` - (Required) The transport protocol used in this rule. Can be one of `tcp` or `udp`.
+
+---
+
+A `node_type` block supports the following:
+
+* `application_port_range` - (Required) Sets the port range available for applications. Format is `<from_port>-<to_port>`, for example `10000-20000`.
+
+* `data_disk_size` - (Required) The size of the data disk in gigabytes..
```
Comment (katbyte) [inline website/docs/r/service_fabric_managed_cluster.html.markdown:154]: would this be better as 
```suggestion
* `data_disk_size_gb` - (Required) The size of the data disk in gigabytes..
```

File: internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go [hunk 4]
```diff
@@ -0,0 +1,1100 @@
...
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+		},
+		"dns_name": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			Computed:     true,
+			ValidateFunc: validation.StringMatch(regexp.MustCompile(`^[a-z0-9]+(-*[a-z0-9])*$`), "The dns name of the cluster must have lowercase letters, numbers and hyphens. The first character must be a letter and the last character a letter or number"),
+		},
+		"dns_service": {
```
Comment (katbyte) [inline internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go:137]:
```suggestion
		"dns_service_enabled": {
```

File: internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go [hunk 5]
```diff
@@ -0,0 +1,1100 @@
...
+		},
+
+		"node_type":      nodeTypeSchema(),
+		"authentication": authSchema(),
+		"custom_fabric_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"parameter": {
```
Comment (katbyte) [inline internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go:177]: I almost feel like this should be 
```suggestion
		"custom_fabric_parameter": {
			Type:     pluginsdk.TypeList,
			Optional: true,
			Elem: &pluginsdk.Resource{
				Schema: map[string]*pluginsdk.Schema{
					"name": {
```

🤔  - WDYT?

File: internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go [hunk 6]
```diff
@@ -0,0 +1,1100 @@
...
+				model.DNSService = true
+			} else if feature == managedcluster.AddonFeaturesBackupRestoreService {
+				model.BackupRestoreService = true
+			}
+		}
+	}
+	model.Username = properties.AdminUserName
+
+	if aad := properties.AzureActiveDirectory; aad != nil {
+
```
Comment (katbyte) [inline internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go:548]:
```suggestion
```

File: website/docs/r/service_fabric_managed_cluster.html.markdown [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+* `vm_instance_count` - (Required) The number of instances this node type will launch.
+
+* `vm_size` - (Required) The size of the instances in this node type.
+
+* `capacities` - (Optional) Specifies a list of key/value pairs used to set capacity tags for this node type.
+
+* `data_disk_type` - (Optional) The type of the disk to use for storing data. It can be one of `Premium_LRS`, `Standard_LRS`, or `StandardSSD_LRS`.
+
+* `multiple_placement_groups` - (Optional) If set the node type can be composed of multiple placement groups.
```
Comment (katbyte) [inline website/docs/r/service_fabric_managed_cluster.html.markdown:176]:
```suggestion
* `multiple_placement_groups_enabled` - (Optional) If set the node type can be composed of multiple placement groups.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @koikonom - overall looks good and I've let some comments inline

**Comment by koikonom** [comment]
(rebased to resolve conflicts)

**Pull request merged**